lib/harmony/resources/resources.go

```diff
@@ -24,6 +24,20 @@ type Resources struct {
 	Gpu       float64
 	Ram       uint64
 	MachineID int
+	Storage
+}
+
+// Optional Storage management.
+// See storagemgr/storagemgt.go for more details.
+type Storage struct {
```

Comment:
I do like how abstract this interface is, probably could be just implemented with the existing storage subsystem.

In .Do how do I get the storage path I'm supposed to use?

Reply:
How about an even more minimal storage interface which allows the task to bring its own storage system:

```go
// technically now this interface would be general enough to just call it "TaskResource"
type Storage interface {
	// taskID may be used to see if some/all of sector files are already in the local storage (possible edge-case when retries of e.g. transfers happen)
	HasCapacity func(taskID int) bool
  
	// Called at the CanAccept scheduling stage, does not have to run on this machine
	Claim func(taskID int) (err error)

	// This allows some other system to consider the task done.
	// It's up to the caller to remove the data, if that applies.
	// May be called on the task which wasn't run in case this node fails to claim it
	MarkComplete func(taskID int) error
}
```

Then it would be really easy to implement exactly this interface as a direct extension to `paths.Local`, and it each task which wants to use the storage we could just roughly:

```go
harmonytask.TaskTypeDetails{
		Max:  s.max,
		Name: "SDR",
		Cost: ...
		// returns harmonytask.Storage
		Storage: task.localStorage.TaskResources(storiface.FTCache, storiface.FSOverheadSeal)
		MaxFailures: 2,
		Follows:     nil,
	}
```

Then in task .Do, like in the sealing pipeline the task would just call a version of `task.localStorage.AcquireSector` with the task ID which would give it the correct paths

Reply:
> how do I get the storage path...
Oops, I thought Do() gets a TaskEngine arg like CanAccept() has. I will need to add it. 
Then one does:
locationsCSV := te.GetWorkingLocation(taskID)
locations = strings.Split(",") // When using storagemgt.CombeFuncs() it will be in the order specified.